Failing tests

Hi @haesleinhuepf @thawn @zoccoler ,

I may need some help here with fixing tests and I'm out of ideas how to fix this. Starting with PR #169, tests have been failing and I have not been able to fix them ever since. Tests are working for me locally.

I have tried several CI changes, which boil down to using wither pytest or tox for the testing.

These are the tracebacks I get from the github CI (tests with conda & tox): 

```Run pytest -v --cov=./ --cov-report=xml
Traceback (most recent call last):
  File "/usr/local/miniconda/envs/test/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 190, in console_main
    code = main()
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 148, in main
    config = _prepareconfig(args, plugins)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 329, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_callers.py", line 55, in _multicall
    gen.send(outcome)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/helpconfig.py", line 103, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1058, in pytest_cmdline_parse
    self.parse(args)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1346, in parse
    self._preparse(args, addopts=addopts)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1229, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest[11](https://github.com/BiAPoL/napari-stress/actions/runs/3321885347/jobs/5490146038#step:8:12)")
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pluggy/_manager.py", line 287, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/usr/local/miniconda/envs/test/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/local/miniconda/envs/test/lib/python3.8/importlib/__init__.py", line [12](https://github.com/BiAPoL/napari-stress/actions/runs/3321885347/jobs/5490146038#step:8:13)7, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 10[14](https://github.com/BiAPoL/napari-stress/actions/runs/3321885347/jobs/5490146038#step:8:15), in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line [16](https://github.com/BiAPoL/napari-stress/actions/runs/3321885347/jobs/5490146038#step:8:17)8, in exec_module
    exec(co, module.__dict__)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pytestqt/plugin.py", line 9, in <module>
    from pytestqt.logging import QtLoggingPlugin, _QtMessageCapture
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 168, in exec_module
    exec(co, module.__dict__)
  File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/pytestqt/logging.py", line 5, in <module>
    from py._code.code import TerminalRepr, ReprFileLocation
ModuleNotFoundError: No module named 'py._code'; 'py' is not a package
Error: Process completed with exit code 1.
```

@haesleinhuepf suggested using the tests from [here](https://github.com/haesleinhuepf/the-segmentation-game/commit/b5cf1ed364fe1ed2e5a86c3637e7cd60a1caefc4) or [here](
https://github.com/haesleinhuepf/napari-skimage-regionprops/commit/1051db2bd4a38b059c2232e27ab53cdd5395f4d4) which I played around with in #181, but to no avail. The error message I get is the same. 

Thanks for any help on this one!

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -32,9 +32,31 @@ jobs:
 
       # these libraries enable testing on Qt on linux
       - uses: tlambert03/setup-qt-libs@v1
-      - name: Test
-        shell: bash -l {0}
-        run: pytest -v --cov=./ --cov-report=xml
+
+      # strategy borrowed from vispy for installing opengl libs on windows
+      - name: Install Windows OpenGL
+        if: runner.os == 'Windows'
+        run: |
+          git clone --depth 1 https://github.com/pyvista/gl-ci-helpers.git
+          powershell gl-ci-helpers/appveyor/install_opengl.ps1
+
+      # note: if you need dependencies from conda, considering using
+      # setup-miniconda: https://github.com/conda-incubator/setup-miniconda
+      # and
+      # tox-conda: https://github.com/tox-dev/tox-conda
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools tox tox-gh-actions
+
+      # this runs the platform-specific tests declared in tox.ini
+      - name: Test with tox
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: python -m tox
+        env:
+          PLATFORM: ${{ matrix.platform }}
+
       - name: Coverage
         uses: codecov/codecov-action@v2
 

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -1,19 +1,16 @@
-# This workflows will upload a Python Package using Twine when a release is created
-# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
-
 name: tests
 
 on:
   push:
     branches:
       - main
-      - master
+      - npe2
     tags:
       - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
   pull_request:
     branches:
-      - main
       - master
+      - main
   workflow_dispatch:
 
 jobs:
@@ -35,36 +32,10 @@ jobs:
 
       # these libraries enable testing on Qt on linux
       - uses: tlambert03/setup-qt-libs@v1
-
-      - name: Set up conda ${{ matrix.python-version }}
-        uses: conda-incubator/setup-miniconda@v2.0.0
-        with:
-          auto-update-conda: true
-          activate-environment: test
-          python-version: ${{ matrix.python-version }}
-          channels: conda-forge
-
-      - name: Conda info
-        shell: bash -l {0}
-        run: conda info
-
-      # note: if you need dependencies from conda, considering using
-      # setup-miniconda: https://github.com/conda-incubator/setup-miniconda
-      # and
-      # tox-conda: https://github.com/tox-dev/tox-conda
-      - name: Install dependencies
-        shell: bash -l {0}
-        run: |
-          python -m pip install --upgrade pip
-          pip install pytest pytest-cov PyQt5 pytest-qt pytest-xvfb
-          pip install -e .
-
-      # this runs the platform-specific tests
-      - name: Run tests
+      - name: Test
         shell: bash -l {0}
         run: pytest -v --cov=./ --cov-report=xml
-
-      - name: Upload coverage to Codecov
+      - name: Coverage
         uses: codecov/codecov-action@v2
 
   deploy:
@@ -87,7 +58,7 @@ jobs:
       - name: Build and publish
         env:
           TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+          TWINE_PASSWORD: ${{ secrets.TWINE_API_KEY }}
         run: |
           git tag
           python -m build .

--- a/.github/workflows/test_and_deploy.yml
+++ b/.github/workflows/test_and_deploy.yml
@@ -47,7 +47,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install setuptools tox tox-gh-actions
+          python -m pip install setuptools tox tox-gh-actions pytest-qt
 
       # this runs the platform-specific tests declared in tox.ini
       - name: Test with tox

--- a/tox.ini
+++ b/tox.ini
@@ -1,13 +1,13 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
-envlist = py{37,38,39}-{linux,macos,windows}
+envlist = py{38,39,310}-{linux,macos,windows}
 isolated_build=true
 
 [gh-actions]
 python =
-    3.7: py37
     3.8: py38
     3.9: py39
+    3.10: py310
     
 [gh-actions:env]
 PLATFORM =

--- a/tox.ini
+++ b/tox.ini
@@ -26,6 +26,14 @@ passenv =
     DISPLAY XAUTHORITY
     NUMPY_EXPERIMENTAL_ARRAY_FUNCTION
     PYVISTA_OFF_SCREEN
-extras =
-    testing
+deps = 
+    pytest  # https://docs.pytest.org/en/latest/contents.html
+    pytest-cov  # https://pytest-cov.readthedocs.io/en/latest/
+    pytest-xvfb ; sys_platform == 'linux'
+    # you can remove these if you don't use them
+    napari
+    magicgui
+    pytest-qt
+    qtpy
+    pyqt5
 commands = pytest -v --color=yes --cov=napari_stress --cov-report=xml

--- a/tox.ini
+++ b/tox.ini
@@ -26,13 +26,6 @@ passenv =
     DISPLAY XAUTHORITY
     NUMPY_EXPERIMENTAL_ARRAY_FUNCTION
     PYVISTA_OFF_SCREEN
-deps = 
-    pytest  # https://docs.pytest.org/en/latest/contents.html
-    pytest-cov  # https://pytest-cov.readthedocs.io/en/latest/
-    # you can remove these if you don't use them
-    napari
-    magicgui
-    pytest-qt
-    qtpy
-    pyqt5
+extras =
+    testing
 commands = pytest -v --color=yes --cov=napari_stress --cov-report=xml

